### DISCUSSION TITLE
Numba, no such module found?

### DISCUSSION THREAD

**Discussion by Ferniclestix** [WASasquatch/was-node-suite-comfyui#87] [General]
installed on a server ive been running things on fine before.

but was having a was nodes failing to load bug. not being a programmer myself I had to monkey around a bit.
Solved by asking my coder buddy XD
make sure your python scripts folder is on the system path. and restart computer. that should fix it more likely than not, you might want to re-update from the requirements.txt too

anyway. maybe this will help someone with a similar bug.

**Comment by WASasquatch** [comment]
There is also a issue, with windows, that if you don't run the requirements explicitly through the python_embedded/python.exe that it will install to the system, and then conflict with Comfy where it can't find the modules, but at the same time, says they're installed. 

To fix that you can uninstall *through* comfy's python.exe and then reinstall.

**Reply by Piscabo** [reply]
Please explain to the rookie how you do this? " uninstall through comfy's python.exe and then reinstall."

**Reply by WASasquatch** [reply]
If you installed to system you can uninstall with confyui's python.exe to remove the conflict if any, and then reinstall specifically with that exe. There are directions on the front page that show how the commands work for running through the python_embedded folders python.exe

**Comment by manny5457** [comment]
![WASinstallerror6](https://github.com/WASasquatch/was-node-suite-comfyui/assets/128967981/ce621760-7b14-46d1-bd45-1888678237ef)
Hi. I'm getting same error "no module named 'numba'". First tried installing via Comfy Manager and it kept failing. Then followed Recommended Installation as per Github page, doing  "pip install -r requirements.txt" from the \custom_nodes\was-node-suite-comfyui folder but failed with same error. I then did it via windows powershell first uninstalling and then installing with "path/to/ComfUI/python_embeded/python.exe -s -m pip install -r requirements.txt" and it did it all, uninstalled and then installed everything and didn't give any errors, but I still get the no module named numba error. I don't know what else I can try.

My python directory is on system path and I'm using windows 10 pro x64. Any help to resolve this would be greatly appreciated, thanks in advance.

**Reply by WASasquatch** [reply]
You would need to edit the path to exactly your python.exe, or it'll be trying to install to the system. You only use pip install by itself you are not using portable and installed to a system python. 

There is also now a install.bat that comes with WAS-NS

**Reply by manny5457** [reply]
Thanks for the help, managed to install it. Uninstalled previous installation then just ran the install.bat you mentioned and its working.

**Comment by Farquaad56** [comment]
same probleme here, search for a tips to fix it thx
`Device: cuda:0 NVIDIA GeForce RTX 2080 Ti
Traceback (most recent call last):
  File "S:\ComfyUI SDXL\ComfyUI\nodes.py", line 1509, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "S:\ComfyUI SDXL\ComfyUI\custom_nodes\was-node-suite-comfyui\__init__.py", line 1, in <module>
    from .WAS_Node_Suite import NODE_CLASS_MAPPINGS
  File "S:\ComfyUI SDXL\ComfyUI\custom_nodes\was-node-suite-comfyui\WAS_Node_Suite.py", line 37, in <module>
    from numba import jit
ModuleNotFoundError: No module named 'numba'

Cannot import S:\ComfyUI SDXL\ComfyUI\custom_nodes\was-node-suite-comfyui module for custom nodes: No module named 'numba'

Import times for custom nodes:
   0.0 seconds: S:\ComfyUI SDXL\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale
   0.0 seconds: S:\ComfyUI SDXL\ComfyUI\custom_nodes\Derfuu_ComfyUI_ModdedNodes
   0.0 seconds (IMPORT FAILED): S:\ComfyUI SDXL\ComfyUI\custom_nodes\was-node-suite-comfyui
   0.3 seconds: S:\ComfyUI SDXL\ComfyUI\custom_nodes\comfyuiManager_v012
   0.7 seconds: S:\ComfyUI SDXL\ComfyUI\custom_nodes\efficiency-nodes-comfyui`

**Comment by WASasquatch** [comment]
Try the install.bat

But it's very important you install to your python embedded. No pip commands by themselves you have to use the python.exe that comes with ComfyUI

**Reply by manny5457** [reply]
I'm the guy above who had the same problem, what you said here "you have to use the python.exe that comes with ComfyUI" this is where I was going wrong, I was giving the location of my locally installed python 6.10 in my user/appdata etc. I didn't realize there was another python installation that came with ComfiUi. Anyway, its working now. Thanks again for your help.
